packages/webview_flutter/webview_flutter_android/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterCookieManager.java

```diff
@@ -37,6 +37,7 @@ void dispose() {
     methodChannel.setMethodCallHandler(null);
   }
 
+  @SuppressWarnings("deprecation")
```

Comment:
Why are you suppressing deprecation warnings for the entire function, rather than just the specific call? If the new code path were to become deprecated, we would want to know.

Reply:
As far as i know it is only possible to suppress the deprecation warning for the whole class or a function, but not a specific call. Please tell me how it's done otherwise.

Reply:
There are a number of examples of statement-level suppressions in the repository: https://github.com/flutter/plugins/search?q=SuppressWarnings

What line or lines are you trying to suppress warnings on, exactly? The issue you've linked from this PR doesn't have any warnings for this method, so it seems like this PR is not for the errors described in the issue.

Reply:
`cookieManager.removeAllCookie();` gives a deprecation warning. I didn't got this warning with my personal project which is where the log is coming from in the issue. I did however get this warning when building the example app. 

I tried suppressing the call only, however it doesn't let me do that. 

![image](https://user-images.githubusercontent.com/20091319/140535650-a0d90d7f-923b-4762-8c32-54cbe1dddf6d.png)

Reply:
@blasten Is there a way to suppress an individual line that doesn't declare a variable?

(Failing that, what we should do is extract the legacy part of the if/else to a dedicated function, and suppress warnings on it.)

Reply:
no, it only works at the class/method level

Reply:
and field level

Reply:
Okay, good to know. In that case we should extract the legacy codepath to a function per my alternate suggestion above.

Reply:
Why not fix the problem rather than hiding it?

https://developer.android.com/reference/android/webkit/CookieManager#public-constructors
says to use this:
https://developer.android.com/reference/android/webkit/CookieManager#getInstance()

Reply:
> Why not fix the problem rather than hiding it?

Supporting older versions of Android is a feature, not a problem. The only problem here is the warning on the legacy branch.

This has in fact already been discussed in this PR.

> https://developer.android.com/reference/android/webkit/CookieManager#public-constructors says to use this: https://developer.android.com/reference/android/webkit/CookieManager#getInstance()

Please read the code in question before leaving review comments.